### DISCUSSION TITLE
[GraphQL] `tokenTransferTxs` incorrect pagination

### DISCUSSION THREAD

**Discussion by kathaypacific** [celo-org/blockscout#912] [Q&A] [answered]
hello,

we're noticing that the `tokenTransferTxs` graphql endpoint for blockscout is returning `hasNextPage: false` in pageInfo unexpectedly. we're relying on this to display the transaction feed in Valora, and the unexpected `hasNextPage` value means that we stop fetching transactions when there are still more pages.

this is the query we are using:
```
query Transfers($address: AddressHash!, $afterCursor: String) {
  # TXs related to cUSD or cGLD transfers
  tokenTransferTxs(addressHash: $address, first: 25, after: $afterCursor) {
    edges {
      node {
        transactionHash
        blockNumber
        timestamp
        gasPrice
        gasUsed
        feeToken
        gatewayFee
        gatewayFeeRecipient
        input
        # Transfers associated with the TX
        tokenTransfer(first: 40) {
          edges {
            node {
              fromAddressHash
              toAddressHash
              fromAccountHash
              toAccountHash
              value
              tokenAddress
              tokenType
              tokenId
            }
          }
        }
      }
    }
    pageInfo {
      startCursor
      endCursor
      hasNextPage
      hasPreviousPage
    }
  }
}
```

an example address this is happening on: `0x2205eacb52c2ddc1518881e06f4e8310056881ef` (although most people can repro using their own addresses also).

unfortunately i'm not sure when this started happening. the last time we worked on the transaction feed data fetching was 10 months ago, at that time we were able to fetch pages until the end of the address history. we haven't really touched this logic since then, i only noticed this issue when testing some unrelated feature today.

let me know if there's some other information i can provide to help debug this :) thanks!

**Comment by arthurgousset** [comment]
Thanks for sharing this here @kathaypacific !

**Comment by rkachowski** [comment]
Hey @kathaypacific can you confirm if this is still an issue? We've changed the currently live version of blockscout which should have addressed this issue.l

**Reply by kathaypacific** [reply, marked as answer]
hi @rkachowski thanks for the quick turnaround, it seems to be working now :)
